### PR TITLE
Added project management commands

### PR DESCRIPTION
**Pull request by e4coder**
## Changelog / Overview

- ``manim init`` - quickly sets up default files for a manim project.
- ``manim new project`` - lets the user set project settings. It also creates the project inside a new folder of name <project_name>
- ``manim new scene`` - used to quickly insert new scenes into files. If ``file name`` is not provided ``main.py`` is used as default.

Added 3 commands in the "manim/cli/project/commands.py" file. Added 2 tests in "tests/test_commands.py"
### Update:
directories have been restructured.
init command resides in manim/cli/init/command.py.
new command group resides in manim/cli/init/group.py

I also have changed the name of manim/cli/cfg/commands.py to manim/cli/cfg/group.py

## Motivation

I want new users to quickly be able to create example scenes and edit those scenes to their liking. Some of my friends don't want to use manim only because they found it very hard

## Explanation for Changes

New users of manim will be able to quickly set up projects and scenes from templates. The number of templates is very few at the moment but we can add more. 
### Commands
1 - manim init : this command quickly sets up default files for a manim project
```
manim init
```
2 - manim new project : this command lets the user set project settings. it also creates the project inside a new folder of name <project_name>. this command can be used in 3 different ways
```
manim new project
manim new project <project_name>
```
3 - manim new scene : this command is used for inserting new scenes in files. if `file name` is not provided `main.py` is used as default. if `file name` is provided but `file name` does not exist. new file is created with name `file name`
```
manim new scene <scene name>
manim new scene <scene name> <file name>
```
## Testing Status

My PC specs:
Processor:............. core I 5 2400
RAM:..................... 8 GB ddr3 ram
GPU:..................... AMD rx 470
Operating System: Ubuntu 20.04

I have tested these commands extensively in ubuntu. I have not tested them in Windows. however i believe they will work fine

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [x] The PR title is descriptive enough
- [x] The PR is labeled correctly
- [x] The changelog entry is completed if necessary
- [x] Newly added functions/classes either have a docstring or are private
- [x] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by e4coder** [comment]
checks are failing for template scenes.

File: manim/cli/project/templates/Graph.py
Comment (kilacoda) [inline manim/cli/project/templates/Graph.py:1]: This is also the name of [another manim class](https://github.com/ManimCommunity/manim/blob/master/manim/mobject/graph.py#L152) which would get overwritten in this case. Might lead to problems if someone tries to insert a `Graph` object here.

Comment (behackl) [inline manim/cli/project/templates/Graph.py:1]: We are in the process of moving away from `GraphScene`. It would be great if this template could also use a construction with `Axes` instead.

File: manim/cli/project/templates/default.py
Comment (kilacoda) [inline manim/cli/project/templates/default.py:1]: This name doesn't conform to [PEP8](https://www.python.org/dev/peps/pep-0008/#class-names)

File: manim/cli/project/templates/MovingCamera.py
Comment (kilacoda) [inline manim/cli/project/templates/MovingCamera.py:1]: [Same as above](https://github.com/ManimCommunity/manim/blob/master/manim/camera/moving_camera.py#L28)

**Review by kilacoda — CHANGES_REQUESTED** [review]

File: manim/cli/project/templates/MovingCamera.py
Comment (e4coder) [reply manim/cli/project/templates/MovingCamera.py:1]: Okay i will change it to MovingCameraTemplate and Graph to GraphTemplate

File: manim/cli/project/templates/default.py
Comment (e4coder) [reply manim/cli/project/templates/default.py:1]: thanks for review i will change it

File: manim/cli/project/templates/Graph.py
Comment (e4coder) [reply manim/cli/project/templates/Graph.py:1]: I'm sorry, do you mean ThreeDAxes or setup_axes?

**Comment by e4coder** [comment]
Requested changes are done

File: manim/cli/project/templates/Graph.py
Comment (jsonvillanueva) [inline manim/cli/project/templates/Graph.py]: This is being deprecated in #1161 which will likely be merged this release cycle. We could use this template, but it will have to be updated; instead, it would make most sense to wait until that PR is merged and use `Axis` instead.

File: manim/cli/project/commands.py
Comment (jsonvillanueva) [inline manim/cli/project/commands.py:18]: I'd greatly prefer if the files structure were made more modular. That is:

- `manim/cli/new/` is its own directory
- `manim/cli/new/commands.py` is its own file that contains the `project`/`scene` decorators
- `manim/cli/init` is its own directory.
- `manim/cli/init/commands.py` is its own file.

I'm okay with the file under `manim/cli/new/` being renamed to `group.py` if you feel that's more appropriate. If that's done, the same should also be done for `manim/cli/cfg`.

File: manim/cli/project/commands.py
Comment (jsonvillanueva) [inline manim/cli/project/commands.py:87]: This content should go under each of the respective helper functions as part of their docstrings. Also, remember capitalization/punctuation.

File: manim/cli/project/commands.py
Comment (jsonvillanueva) [inline manim/cli/project/commands.py:95]: We use NumPy format for our docs, see how that would be used here: https://docs.manim.community/en/stable/contributing/documentation.html#numpy-format

File: manim/cli/project/commands.py
Comment (jsonvillanueva) [inline manim/cli/project/commands.py:284]: Remember punctuation in the command/option help tags

File: manim/cli/project/templates/Graph.mtp
Comment (jsonvillanueva) [inline manim/cli/project/templates/Graph.mtp:1]: We are deprecating GraphScene in #1161 soon and it would not be great to continue its usage in a template which should encourage best (new) practices. In which case, GraphTemplate should really be an AxisTemplate once #1161 is merged.

**Review by jsonvillanueva — CHANGES_REQUESTED** [review]
Looks good as a first passover. I'll have to rereview after refactors/restructure.

File: manim/cli/project/templates/Graph.py
Comment (e4coder) [reply manim/cli/project/templates/Graph.py]: yes, it is manim template. Thanks for the review. I will remove this template for now and add another one.

File: manim/cli/project/commands.py
Comment (e4coder) [reply manim/cli/project/commands.py:18]: Okay i can do that. that would be easy to navigate

File: manim/cli/project/commands.py
Comment (e4coder) [reply manim/cli/project/commands.py:87]: okay thanks again for the review. i will start working on it

File: manim/cli/project/templates/Graph.py
Comment (e4coder) [reply manim/cli/project/templates/Graph.py]: After reading #1161 I now understand what Axis means. I will switch the graph scene with Axis.

**Comment by e4coder** [comment]
I have refactored the code, fixed docstrings, added AxesTemplate instead of GraphTemplate, and restructured the directories

File: manim/cli/project/templates/Graph.mtp
Comment (e4coder) [reply manim/cli/project/templates/Graph.mtp]: Done

File: manim/cli/project/commands.py
Comment (e4coder) [reply manim/cli/project/commands.py]: Done

File: manim/cli/project/commands.py
Comment (e4coder) [reply manim/cli/project/commands.py]: done

File: manim/cli/project/commands.py
Comment (e4coder) [reply manim/cli/project/commands.py]: Done

File: manim/cli/project/templates/Graph.py
Comment (e4coder) [reply manim/cli/project/templates/Graph.py]: I am sorry i did not know that GraphScene was being deprecated. I removed GraphTemplate and added AxesTemplate

Comment (behackl) [reply manim/cli/project/templates/Graph.py]: There was no way for you to know, don't worry about it. :-)

File: manim/cli/new/group.py
Comment (jsonvillanueva) [inline manim/cli/new/group.py:24]: We can figure this out ourselves via search if we're curious
```suggestion
```

File: manim/cli/new/group.py
Comment (jsonvillanueva) [inline manim/cli/new/group.py:74]:
```suggestion
```

File: manim/cli/new/group.py
Comment (jsonvillanueva) [inline manim/cli/new/group.py:89]: I'm not sure why `'new folder'` is in quotes here. It seems like this should be `'project_name'` instead since that Path variable would be the folder created?

File: manim/cli/new/group.py
Comment (jsonvillanueva) [inline manim/cli/new/group.py:147]: I would use this help message in the docstring down below instead of here. See render command and how it's done there.
```suggestion
    help="Insert a scene to an existing file or a new file.",
```

File: manim/cli/new/group.py
Comment (jsonvillanueva) [inline manim/cli/new/group.py:152]: Same as in other cases, split into separate lines. See `cli/render/commands.py`'s `render()` command as an example. It's good to use this docstring space to document the command's positional arguments .

File: manim/cli/init/commands.py
Comment (jsonvillanueva) [inline manim/cli/init/commands.py:20]: This is too long currently. Check with `manim -h`

Comment (jsonvillanueva) [inline manim/cli/init/commands.py:20]: Using this `help` option overwrites the content of the docstring down below. You should prefer the docstring down below so that you can be more descriptive in the help message.

File: manim/cli/new/group.py
Comment (jsonvillanueva) [inline manim/cli/new/group.py:157]: I don't think this is necesesary (as well as the other instances), but if you'd really like to keep it whilst using the docstring for the help message, there's a click formatting string to bypass outputting to the help page. using \f

See the docs for click here:
https://click.palletsprojects.com/en/7.x/documentation/#truncating-help-texts

**Review by jsonvillanueva — COMMENTED** [review]
This is coming together quite well. I've left a few more comments to address; though I'd like to checkout the branch and test it out before approving... but that'll have to wait until tomorrow.

File: manim/cli/new/group.py
Comment (e4coder) [reply manim/cli/new/group.py:157]: I will try to do everything before the day ends. thanks for the review.

File: manim/cli/new/group.py
Comment (e4coder) [reply manim/cli/new/group.py:89]: yes totally. my bad sorry. I changed it to project_name. I am not very good at naming stuff.

**Comment by e4coder** [comment]
I think I touched everything mentioned in the comments. let me know if I missed anything. thx

File: manim/cli/init/commands.py
Comment (e4coder) [reply manim/cli/init/commands.py]: I am sorry i did not realize the 'Commit suggestion' button would automatically commit the suggested changes. however i did it manually.

File: manim/cli/init/commands.py
Comment (jsonvillanueva) [inline manim/cli/init/commands.py:21]:
```suggestion
    """Sets up a project in current working directory.
```

File: manim/cli/new/group.py
Comment (jsonvillanueva) [inline manim/cli/new/group.py:136]:
```suggestion
    """Inserts a SCENE to an existing FILE or creates a new FILE.
```

**Review by jsonvillanueva — CHANGES_REQUESTED** [review]
I think these are the last tidbits for making the CLI display help pages properly. Other than this, I think the subcommands work as intended. It may be worth adding documentation for these subcommands somewhere in the [tutorials section](https://docs.manim.community/en/latest/tutorials.html), but I think this can be done in a future PR.

File: manim/cli/new/group.py
Comment (e4coder) [reply manim/cli/new/group.py]: I was planning on adding documentation for these in the future.

thanks a bunch for the review

**Review by jsonvillanueva — APPROVED** [review]

**Comment by e4coder** [comment]
@kilacoda Sir it's been 4 days since you requested changes. could you possibly review them? thx

**Comment by kilacoda** [comment]
Sorry, got caught up in some other stuff! Reviewing rn.

File: manim/cli/init/commands.py
Comment (kilacoda) [inline manim/cli/init/commands.py:21]: Since this also appears to be the help string for this command, maybe it'd be better to specify that the project is created with default settings?
```suggestion
    """Sets up a project in current working directory with default settings.
```

another option might be to explicitly add a `help` argument to the decorator (like with `new`)

Comment (e4coder) [reply manim/cli/init/commands.py]: I used the help argument with the decorator but, jsonvilanueva said that click uses docstrings as help.

Comment (e4coder) [reply manim/cli/init/commands.py]: if i type manim init -h it shows the docstring

Comment (e4coder) [reply manim/cli/init/commands.py]: sorry, my English is bad.

**Review by kilacoda — APPROVED** [review]
LGTM!

**Pull request merged**
